src/program/services/streaming/media_stream.py

```diff
@@ -126,12 +127,34 @@ def __init__(
                 )
             )
 
-        try:
-            self.async_client = di[httpx.AsyncClient]
-        except KeyError:
-            raise RuntimeError(
-                "httpx.AsyncClient not found in dependency injector"
-            ) from None
+        # Use proxy for AllDebrid if configured
+        if provider == "alldebrid":
+            proxy_url = settings_manager.settings.downloaders.proxy_url
+            if proxy_url:
+                logger.log(
+                    "STREAM",
+                    self._build_log_message(
+                        f"Using proxy {proxy_url} for AllDebrid streaming"
+                    ),
+                )
+                # Create a dedicated client with proxy for this stream
+                mounts = {
+                    "http://": httpx.AsyncHTTPTransport(proxy=proxy_url),
+                    "https://": httpx.AsyncHTTPTransport(proxy=proxy_url),
+                }
+                self.async_client = httpx.AsyncClient(http2=True, mounts=mounts)
```

Comment:
Do we _need_ to create a client for each stream, or can it be one client per proxy?

HTTPX suggest using a single client (it reuses resources and connections), so I'd like to avoid spinning up lots of new ones where possible 😄